site/app/views/forum/ForumThreadView.php

```diff
@@ -755,13 +757,14 @@ public function displayThreadList($threads, $filtering, &$activeThreadAnnounceme
         foreach ($author_user_groups as $author) {
             $is_instructor_full_access[$author["user_id"]] = $author["user_group"] <= User::GROUP_FULL_ACCESS_GRADER;
         }
+        $first_posts = $this->core->getQueries()->getFirstPostForThreads(array_column($threads, "id"));
 
         foreach ($threads as $thread) {
             // Checks if thread ID is empty. If so, skip this threads.
             if (empty($thread["id"])) {
                 continue;
             }
-            $first_post = $this->core->getQueries()->getFirstPostForThread($thread["id"]);
+            $first_post = (isset($first_posts[$thread["id"]])) ? $first_posts[$thread['id']] : null;
```

Comment:
This is a great chance to use the [null coalescing](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce) operator!

```suggestion
            $first_post = $first_posts[$thread["id"]] ?? null;
```